### ISSUE TITLE
Add `ImplicitArray`

### ISSUE THREAD

**Issue by czgdp1807** [codezonediitj/pydatastructs#457]
#### Description of the problem

As of now in `OneDimensionalArray` and subclasses we store all the data explicitly in memory. `ImplicitArray` will just store a function object. This function object will return values for different indices. That way the memory requirement will be reduced to a constant. This will be useful for cases where search algorithms don't need an explicit array but a range and some values associated with that range.

**Comment by HassanRahim26** [comment]
Hey, @czgdp1807 I want to work on this issue. Can you please tell me ways to get started?

**Comment by czgdp1807** [comment]
Sure go ahead. Well it's about adding new classes (`ImplicitArray`, `OneDimensionalImplicitArray`) in `arrays.py` with same interface (methods in simpler terms) as `Array`, `OneDimensionalArray`. There is a lot of code in `arrays.py` which you can go through and follow the same pattern for adding implicit arrays.

**Comment by HassanRahim26** [comment]
So, do I've to create a new class like this:-

```py
class ImplicitArray(Array):
    pass

def OneDimensionalImplicitArray(ImplicitArray, OneDimensionalArray):
    "Here I've to declare the functions of this class"
```

**Comment by czgdp1807** [comment]
Yeah exactly. In fact, the classes you have inherited seems about right.

**Comment by HassanRahim26** [comment]
Alright then. I'll work on this issue and I'll create a PR for this.

**Comment by NIveX2169** [comment]
hi can you assign me this issue under GSSOC23 .

**Comment by Vraj2811** [comment]
Hi @czgdp1807, is the issue still open?
Can I work on it ?

**Comment by FamALouiz** [comment]
I believe this is still open. Will give it a shot.
